dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/program/DefaultProgramIndicatorService.java

```diff
@@ -126,7 +126,7 @@ public DefaultProgramIndicatorService(
     this.analyticsSqlCache = cacheProvider.createAnalyticsSqlCache();
     this.sqlBuilder = sqlBuilder;
 
-    this.programIndicatorItems = new ExpressionMapBuilder(sqlBuilder).getExpressionItemMap();
+    this.programIndicatorItems = new ExpressionMapBuilder().getExpressionItemMap();
```

Comment:
I’m curious — was there a particular reason we opted not to use dependency injection in this case?

Reply:
I think it's OK. In `DefaultExpressionService` these maps are still generated in the service (for all the expression types that are not program indicators), so this is not consistent with that. But all this ANLR code will be ripped out when we transition to Jan B's new parser. (Did you know about that?) I'm OK with keeping it as a separate class or putting it back where it was.